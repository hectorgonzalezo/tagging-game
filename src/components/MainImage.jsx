import React, { useState } from 'react';
import comiconImg from '../assests/comicon.jpg';
import Target from './Target';
import '../styles/mainImageStyle.css';

function MainImage() {
  // This will have the targets chosen by user, both wrong and right
  const [targets, setTarget] = useState([]);

  // This function allow the user to click on a location in image
  function clickLocation(e) {
    console.log(e)
    const { pageX, pageY } = e;
    setTarget([...targets, { x: pageX, y: pageY }]);
  }
  return (
    <div id="img-container">
      <img src={comiconImg} alt="" id="main-image" onClick={clickLocation}/>
      {targets.map((target) => <Target key={`${target.x}-${target.y}`} location={target} />)}
    </div>
  );
}

export default MainImage;
