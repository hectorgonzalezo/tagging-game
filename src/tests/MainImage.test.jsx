import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import userEvent from '@testing-library/user-event';
import MainImage from '../components/MainImage';

describe('Database queries', () => {
  test('Pressing anywhere adds a target', async () => {
    render(<MainImage />);
    const image = screen.getByRole('img');

    userEvent.click(image);

    // After click there should be a target
    const target = screen.getAllByRole('generic');
    expect(target[2]).toHaveClass('target');
  });

  test('Adds multiple targets', async () => {
    render(<MainImage />);
    const image = screen.getByRole('img');

    userEvent.click(image);
    userEvent.click(image, { pageX: 10, pageY: 20 });
    userEvent.click(image, { pageX: 50, pageY: 600 });

    const target = screen.getAllByRole('generic');
    // There should be 5 divs after three clicks
    expect(target.length).toBe(8);
    expect(image).toMatchSnapshot();
  });
});
