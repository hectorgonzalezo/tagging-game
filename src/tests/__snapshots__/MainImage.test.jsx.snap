// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Database queries Adds multiple targets 1`] = `
<img
  alt="main"
  id="main-image"
  src="comicon.jpg"
/>
`;
